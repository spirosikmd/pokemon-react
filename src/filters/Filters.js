import React, { PureComponent } from 'react';
import { Input } from 'semantic-ui-react';

class Filters extends PureComponent {
  constructor(props) {
    super(props);

    this.handleSearchTextInputChange = this.handleSearchTextInputChange.bind(
      this
    );
    this.handleMyPokemonInputChange = this.handleMyPokemonInputChange.bind(
      this
    );
  }

  handleSearchTextInputChange(event) {
    this.props.onSearchTextInput(event.target.value);
  }

  handleMyPokemonInputChange(event) {
    this.props.onMyPokemonInput(event.target.checked);
  }

  render() {
    return (
      <form>
        <label>
          <Input
            type="text"
            placeholder="Search..."
            value={this.props.searchText}
            onChange={this.handleSearchTextInputChange}
          />
        </label>
        <label>
          <Input
            type="checkbox"
            checked={this.props.myPokemonOnly}
            onChange={this.handleMyPokemonInputChange}
          />
          Only show my pokemon
        </label>
      </form>
    );
  }
}

export default Filters;
