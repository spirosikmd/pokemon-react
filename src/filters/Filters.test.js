import React from 'react';
import renderer from 'react-test-renderer';
import Filters from './Filters';

const mockHandleSearchTextInput = jest.fn();
const mockHandleOnFavoriteInput = jest.fn();

test('Filters renders filters and handle data change', () => {
  const component = renderer.create(
    <Filters
      searchText="pickachu"
      favoriteOnly={true}
      onSearchTextInput={mockHandleSearchTextInput}
      onFavoriteInput={mockHandleOnFavoriteInput}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const searchTextInput = tree.children[0].children[0];
  searchTextInput.props.onChange({
    target: {
      value: 'bulbasar',
    },
  });
  expect(mockHandleSearchTextInput.mock.calls.length).toBe(1);
  expect(mockHandleSearchTextInput.mock.calls[0][0]).toBe('bulbasar');

  const favoriteOnlyCheckboxInput = tree.children[1].children[0].children[0];
  favoriteOnlyCheckboxInput.props.onChange({
    target: {
      checked: false,
    },
  });
  expect(mockHandleOnFavoriteInput.mock.calls.length).toBe(1);
  expect(mockHandleOnFavoriteInput.mock.calls[0][0]).toBe(false);
});
