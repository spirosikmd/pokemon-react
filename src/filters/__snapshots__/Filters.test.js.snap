exports[`test Filters renders filters and handle data change 1`] = `
<form>
  <label>
    <input
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value="pickachu" />
  </label>
  <p>
    <label>
      <input
        checked={true}
        onChange={[Function]}
        type="checkbox" />
      Only show favorite pokemon
    </label>
  </p>
</form>
`;
